lazy_theta_star_b/src/lazy_theta_star_b.cpp

```diff
@@ -0,0 +1,529 @@
+    }
+
+    void LazyThetaStarB::deactivate() {
+        RCLCPP_INFO(
+                node_->get_logger(), "Deactivating plugin %s of type LazyThetaStarB",
+                name_.c_str());
+    }
+
+    nav_msgs::msg::Path LazyThetaStarB::createPlan(
+            const geometry_msgs::msg::PoseStamped &start,
+            const geometry_msgs::msg::PoseStamped &goal) {
+
+
+           RCLCPP_INFO(
+                node_->get_logger(), "GOT THE SOURCE AND DESTINATION ------ (%f, %f) && (%f, %f)",
+               start.pose.position.x, start.pose.position.y, goal.pose.position.x, goal.pose.position.y);
+
+        costmap_->worldToMap(start.pose.position.x, start.pose.position.y, reinterpret_cast<unsigned int &>(src.x),
+                             reinterpret_cast<unsigned int &>(src.y));
+
+
+        costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, reinterpret_cast<unsigned int &>(dst.x),
+                             reinterpret_cast<unsigned int &>(dst.y));
+        src.x = int(src.x);
+        src.y = int(src.y);
+
+        dst.x = int(dst.x);
+        dst.y = int(dst.y);
+
+        data.clear();
+        pq.clear();
+        path.clear();
+        posn.clear();
+        lethal_cost = LETHAL_COST;
+
+        sizeX = costmap_->getSizeInCellsX();
+        sizeY = costmap_->getSizeInCellsY();
+
+        //has been added so that if the robot is stuck at a place with costmap cost higher than 127, then only for that run, the planner be able to give a path
+        if (costmap_->getCost(src.x, src.y) >= lethal_cost && costmap_->getCost(src.x, src.y) <= 253) {
+            lethal_cost = costmap_->getCost(src.x, src.y);
+        }
+        
+        initializePosn();
+        data.reserve(int((sizeX) * (sizeY) * 0.01));
+
+        clearRobotCell(src.x, src.y);
+
+         if (!isSafe(src.x, src.y) || !isSafe(dst.x, dst.y) || !withinLimits(src.x, src.y) ||
+            !withinLimits(dst.x, dst.y)) {
+            RCLCPP_INFO(node_->get_logger(), "THE SOURCE OR DESTINATION IS EITHER NOT WITHIN THE SIZE MAP OR IS OBSTRUCTED");
+            global_path.poses.clear();
+            return global_path;
+
+        }
+
+        data.push_back({(src.x), (src.y), 0, dist(src.x, src.y, dst.x, dst.y), 0, -1,
+                        dist(src.x, src.y, dst.x, dst.y)});
+
+        addIndex(src.x, src.y, 0);
+
+        pq.push_back({0, &(data[0].g), &(data[0].f)});
+        //added as a buffer, since my binary_heap_del_min starts from the index 1 instead of 0
+        pq.push_back({0, &(data[0].g), &(data[0].f)});
+
+
+        //The algorithm begins here
+        id curr_id = 0;
+        id id_gen = 1;
+        id last_id = 0;
+        int cy;
+        int cx;
+        cx = (data[curr_id].x);
+        cy = (data[curr_id].y);
+
+        if (!los_check(src.x, src.y, dst.x, dst.y)) {
+            
+            while (pq.size() > 1) {
+
+                data[curr_id].counter++;
+               
+                 int moves[8][2] = {{cx,     cy + 1},
```

Comment:
Allocating this every iteration is costly. You could store instead a `base_moves` variable with just the +/- 1 that you add to the current `cx`, so you'd do `cx + moves[i]` so that you don't reallocate this off the stack every loop